cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/system/SystemWorkerHttpHandlerInternal.java

```diff
@@ -90,8 +92,10 @@
 
   private final MetricsCollectionService metricsCollectionService;
 
-  public SystemWorkerHttpHandlerInternal(CConfiguration cConf, MetricsCollectionService metricsCollectionService) {
-    this.runnableTaskLauncher = new RunnableTaskLauncher(cConf);
+  public SystemWorkerHttpHandlerInternal(CConfiguration cConf, MetricsCollectionService metricsCollectionService,
+    Injector injector) {
+    this.runnableTaskLauncher = new RunnableTaskLauncher(cConf, injector);
```

Comment:
it's kind of weird to pass an injector around, any reason we can't pass the RunnableTaskLauncher in here?

Reply:
The need for passing injector was discussed offline.